### DISCUSSION TITLE
leetcode/283

### DISCUSSION THREAD

**Discussion by giscus** [fuxuemingzhu/Leetcode-Solution-All#14] [General]
# leetcode/283

负雪明烛的刷题博客，LeetCode算法题解、刷题模板

https://fuxuemingzhu.cn/leetcode/283.html

**Comment by aamada** [comment]
int left = 0;
        int right = 0;
        while (right < nums.length) {
            if (nums[right] == 0) {
                right++;
                continue;
            }
            if (left != right) {
                // 交换
                nums[left] = nums[right];
                nums[right] = 0;
                left ++;
                right++;
                continue;
            }
            left ++;
            right++;
        }

**Reply by dfsj66011** [reply]
```python
slow, quick = 0, 0

while quick < len_nums - 1:
    while nums[quick] == 0 and quick < len_nums - 1:
        quick += 1
    while slow <= quick and nums[slow] != 0:  
        slow += 1
    nums[slow], nums[quick] = nums[quick], nums[slow]
return nums
```
